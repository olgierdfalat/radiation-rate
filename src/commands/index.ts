export * from './checksum';
export * from './filterFiles';
export * from './wrongFiles';
export * from './excelExport';
