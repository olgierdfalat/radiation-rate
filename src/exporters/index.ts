export * from './worksheetsDataProvider';
export * from './excel';