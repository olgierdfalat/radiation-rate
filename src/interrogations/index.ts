export * from './interrogation';
export * from './stJude';
export * from './biotronikStd';
export * from '../exporters/worksheetsDataProvider';
export * from './interrogationFactory';