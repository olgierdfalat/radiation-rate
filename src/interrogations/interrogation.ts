import * as fs from 'fs';
import * as models from './../models';
import { promisify } from 'util';
import yaml from 'js-yaml';

const readFile = promisify(fs.readFile);


export class Interrogation {
  private fileName: string;
  constructor(fileName: string) {
    this.fileName = fileName;
  }

  async getContent(): Promise<string> {
    return (await readFile(this.fileName)).toString();
  }

  async getData(): Promise<models.StJudeRowModel> {
    throw new Error('Not supported');
  }

  async getAsYaml() {
    return yaml.dump(await this.getData());
  }
}
