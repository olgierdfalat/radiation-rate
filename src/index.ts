import { Command } from 'commander';
import * as commands from './commands';


const command = new Command();
command
  .version('0.0.1')
  .description('CLI for converting heart valves output data into Excel format')
  .option('-c, --checksum <device>', 'generates checksums of all log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.checksum)
  .option('-e, --excluded <device>', 'generates list of excluded log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.excludedFiles)
  .option('-i, --included <device>', 'generates list of included log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.includedFiles)
  .option('-w, --invalid <device>', 'generates list of wrong log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.invalidFiles)
  .parse(process.argv);

