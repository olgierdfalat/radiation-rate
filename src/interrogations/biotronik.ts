import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class Biotronik extends Interrogation {
  async getData(): Promise<parsers.BiotronikData> {
    const content = await this.getContent();
    return new parsers.BiotronikData(content, this.filePath);
  }
}