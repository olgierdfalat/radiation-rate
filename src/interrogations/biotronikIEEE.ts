import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikIEEE extends Interrogation {
  async getData(): Promise<parsers.BiotronikIEEEData> {
    const content = await this.getContent();
    return new parsers.BiotronikIEEEData(content, this.filePath);
  }
}