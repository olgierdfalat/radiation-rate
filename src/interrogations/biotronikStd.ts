import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikStd extends Interrogation {
  async getData(): Promise<parsers.BiotronikStdData> {
    const content = await this.getContent();
    return new parsers.BiotronikStdData(content, this.filePath);
  }
}