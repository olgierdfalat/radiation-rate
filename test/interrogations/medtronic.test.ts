import * as path from 'path';
import * as interrogations from '../../src/interrogations';

describe('interrogation Medtronic class', () => {
  it('should get structured data row from file', async () => {
    const filePath = path.join(__dirname, '../../../radiation-rate-data/test/Medtronic/PYZ626053S_17123650.xls');
    const medtronic = new interrogations.Medtronic(filePath);
    const data = await medtronic.getData();
    expect(data.getRow()).toMatchSnapshot();
  });
});