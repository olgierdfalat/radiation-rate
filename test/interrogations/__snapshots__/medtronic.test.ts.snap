// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interrogation Medtronic class should get structured data row from file 1`] = `
Array [
  Object {
    "name": "ToolGenerator",
    "type": "string",
    "value": "PDDTOOLS (C) 2007 Medtronic Inc.",
  },
  Object {
    "name": "Version",
    "type": "string",
    "value": "Rev. 2.005 23-Aug-13",
  },
  Object {
    "name": "Note",
    "type": "string",
    "value": "*** FOR INTERNAL USE ONLY ***",
  },
  Object {
    "name": "LIA RAMware Status",
    "type": "string",
    "value": "INSTALLED",
  },
  Object {
    "name": "Model Identification:",
    "type": "string",
    "value": "Consulta CRT-D D234TRK",
  },
  Object {
    "name": "Serial Number:",
    "type": "string",
    "value": "PYZ626053S",
  },
  Object {
    "name": "Binary File:",
    "type": "string",
    "value": "17123650.pdd",
  },
  Object {
    "name": "Binary File Saved on:",
    "type": "string",
    "value": "17-Apr-15 12:36:50",
  },
  Object {
    "name": "Open XML-file for more info",
    "type": "string",
    "value": "17123650.xml",
  },
  Object {
    "name": "Open Waveform Viewer",
    "type": "string",
    "value": "Waveform Viewer",
  },
  Object {
    "name": "Prior to Last Session Timestamp",
    "type": "date",
    "value": "15-Apr-2015 14:44:59",
  },
  Object {
    "name": "Last Session Timestamp",
    "type": "date",
    "value": "16-Apr-2015 14:56:33",
  },
  Object {
    "name": "Last Clear Timestamp",
    "type": "date",
    "value": "23-Jun-2014 14:32:43",
  },
  Object {
    "name": "Last Remote Interrogation Timestamp",
    "type": "date",
    "value": "N/A",
  },
  Object {
    "name": "Device Implant Timestamp",
    "type": "date",
    "value": "23-Jun-2014 14:32:43",
  },
  Object {
    "name": "Last Interrogated Timestamp",
    "type": "date",
    "value": "17-Apr-2015 13:33:08",
  },
  Object {
    "name": "Days Since Last Session",
    "type": "string",
    "value": "0.94",
  },
  Object {
    "name": "Days in Previous Session",
    "type": "string",
    "value": "1.47",
  },
  Object {
    "name": "Audit Rule(s)/Observations:",
    "type": "string",
    "value": null,
  },
  Object {
    "name": "Vf Dx Off",
    "type": "string",
    "value": "",
  },
  Object {
    "name": "Alert Defib Lead Warning",
    "type": "string",
    "value": "Date Time, 16-Apr-2015 15:00:22",
  },
  Object {
    "name": "Alert SVC Lead Warning",
    "type": "string",
    "value": "Date Time, 16-Apr-2015 15:00:23",
  },
  Object {
    "name": "Alert V Bipolar Lead Warning",
    "type": "string",
    "value": "Date Time, 16-Apr-2015 15:00:10",
  },
  Object {
    "name": "Alert A Bipolar Lead Warning",
    "type": "string",
    "value": "Date Time, 16-Apr-2015 15:00:04",
  },
  Object {
    "name": "Alert LV Perm Pacing Polarity Lead Warning",
    "type": "string",
    "value": "Polarity, L Vtip To R Vcoil, Date Time, 16-Apr-2015 15:00:18",
  },
  Object {
    "name": "LVCM Cannot Measure LV Threshold Last 7 Days",
    "type": "string",
    "value": "",
  },
  Object {
    "name": "FDI Small",
    "type": "string",
    "value": "",
  },
  Object {
    "name": "Vt Dx Off",
    "type": "string",
    "value": "",
  },
  Object {
    "name": "DOOMode",
    "type": "string",
    "value": "",
  },
  Object {
    "name": "Time Of Last Battery Measurement",
    "type": "string",
    "value": "17-Apr-2015 13:33:04",
  },
  Object {
    "name": "Last Battery Measurement",
    "type": "string",
    "value": "3.0187 V, ERI/RRT=2.6251 V, EOL=2.5534 V",
  },
  Object {
    "name": "Time of RRT",
    "type": "string",
    "value": "N/A",
  },
]
`;
